src/views/Home.vue

```diff
@@ -0,0 +1,398 @@
+            group.selectedShipmentMethodTypeId = group.shipmentMethodTypeId;
+            group.items.map((item: any) => {
+              if (item.productId) productIds.add(item.productId);
+              if (item.status == 'ITEM_CANCELLED') isShipmentCancelled = true;
+            })
+            group.isShipmentCancelled = isShipmentCancelled;
+          })
+          this.productIds = [...productIds]
+          await this.fetchProducts(this.productIds)
+        } else {
+          this.order = {}
+          showToast(translate("Order not Found"))
+        }
+      } catch (error) {
+        this.order = {}
+        console.error(error)
+        showToast(translate("Something went wrong while fetching the order details"))
+      }
+    },
+
+    async fetchProducts(productIds: any) {
+      const productIdFilter = productIds.reduce((filter: string, productId: any) => {
+        if (filter !== '') filter += ' OR '
+        return filter += productId;
+      }, '');
+
+      try {
+        const resp = await ProductService.fetchProducts({
+          "filters": ['productId: (' + productIdFilter + ')'],
+          "viewSize": productIds.length
+        })
+
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          resp.data.response.docs.forEach((product: any) => {
+            this.products[product.productId] = product
+          });
+        }
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong"))
+      }
+    },
+
+    getProduct(productId: string) {
+      return this.products[productId] ? this.products[productId] : {}
+    },
+
+    async saveShippingAddressChanges(shipGroup: any) {
+      let resp
+      const payload = {
+        "orderId": this.order.id,
+        "shipGroupSeqId": shipGroup.shipGroupSeqId,
+        "contactMechId": shipGroup.shipTo.postalAddress.id,
+        "shipmentMethod": `${this.deliveryMethod}@_NA_`,
+        "contactMechPurposeTypeId": "SHIPPING_LOCATION",
+        "facilityId": "WH",
+        "toName": `${shipGroup.editedShipmentAddress.firstName} ${shipGroup.editedShipmentAddress.lastName}`,
+        "address1": shipGroup.editedShipmentAddress.address1,
+        "city": shipGroup.editedShipmentAddress.city,
+        "stateCode": shipGroup.editedShipmentAddress.stateCode,
+        "postalCode": shipGroup.editedShipmentAddress.postalCode,
+        "country": shipGroup.editedShipmentAddress.country
+      } as any
+
+      if (shipGroup.selectedShipmentMethodTypeId === shipGroup.shipmentMethodTypeId) {
+        // In case of address edit, we honour the previously selected delivery method
+        payload.shipmentMethod = `${shipGroup.shipmentMethodTypeId}@_NA_`
+        payload.isEdited = true
+      }
+
+      try {
+        resp = await OrderService.updateShippingAddress(payload);
+        if (resp.status === 200 && !hasError(resp) && resp.data) {
+          shipGroup.areChangesSaved = true;
```

Comment:
I think this is not needed, we could replace address with updated one and remove editedAddress/updatedAddress field